peps/pep-0752.rst

```diff
@@ -0,0 +1,399 @@
+__ https://packaging.python.org/en/latest/specifications/name-normalization/#name-normalization
+
+Grant Semantics
+---------------
+
+A namespace grant bestows ownership over the following:
+
+1. A package matching the namespace itself such as the placeholder package
+   `microsoft <https://pypi.org/project/microsoft/>`__.
+2. Packages that start with the namespace followed by a hyphen. For example,
+   the namespace ``foo`` would match the package ``foo-bar`` but not the
+   package ``foobar``.
+
+Grant Types
+-----------
+
+There are two types of grants.
+
+.. _root-grant:
+
+Root Grant
+''''''''''
+
+Only organizations have the ability to submit requests for namespace grants.
+An organization gets a root grant for every accepted request. This grant may
+produce any number of `child grants <child-grant_>`_.
+
+.. _child-grant:
+
+Child Grant
+'''''''''''
+
+A child grant is created by the owner of a `root grant <root-grant_>`_. The
+namespace MUST be prefixed by the root grant namespace followed by a hyphen.
+For example, ``google-cloud`` would be a valid child of the root ``google``.
+
+.. _grant-ownership:
+
+Grant Ownership
+---------------
+
+The owner of a grant may allow the grant to be shared with any number of other
+organizations. Organizations with which grants are shared cannot themselves
+share such grants. For example, if organization A was given a root grant and
+decides to share this with organization B:
+
+* Organization B cannot share the grant with other organizations.
+* Organization B cannot produce a child grant.
+
+The owner may also transfer this executive ownership to another organization.
+If the organization is a corporate organization, the target for transfer must
+also be.
+
+.. _uploads:
+
+Uploads
+-------
+
+If the following criteria are all true for a given upload:
+
+1. The package does not yet exist.
+2. The name matches a reserved namespace.
+3. The user is not authorized to use the namespace by the owner of the
+   namespace.
+
+Then the upload MUST fail with a 403 HTTP status code.
+
+.. _user-interface:
+
+User Interface
+--------------
+
+Every page for a particular release
+(`example <https://pypi.org/project/google-cloud-compute/1.19.2/>`__)
+that both matches an active namespace grant and is tied to an
+`owner <grant-ownership_>`_
+MUST receive a special indicator that signifies this tie. The UI also MUST
+indicate what the prefix is (NuGet does not do this). Repositories SHOULD have
+a dedicated page that enumerates every active namespace grant and which
+organization(s) own it.
+
+.. _public-namespaces:
+
+Public Namespaces
+-----------------
+
+The owner of a grant may choose to allow others the ability to release new
+packages with the associated namespace. Doing so MUST allow
+`uploads <uploads_>`_ for new packages matching the namespace from any user
+but such releases MUST NOT have the `visual indicator <user-interface_>`_.
+
+Root grants given to community projects SHALL always be public.
```

Comment:
Can this section talk about the interaction of public namespaces with shared ownership? Both mean that more than one entity can upload packages, but have very different semantics so we should point this out

Reply:
I added:

```
It is possible for the `owner <grant-ownership_>`_ of a namespace to both make it public and allow other organizations to use it. In this case, the permitted organizations have no special permissions and are essentially only public.
```